cmd/update_prs/update_prs.go

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2021 Skyscanner Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package update_prs
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/skyscanner/turbolift/internal/campaign"
+	"github.com/skyscanner/turbolift/internal/colors"
+	"github.com/skyscanner/turbolift/internal/github"
+	"github.com/skyscanner/turbolift/internal/logging"
+	"github.com/skyscanner/turbolift/internal/prompt"
+	"github.com/spf13/cobra"
+)
+
+var (
+	gh github.GitHub = github.NewRealGitHub()
+	p  prompt.Prompt = prompt.NewRealPrompt()
+)
+
+var (
+	closeFlag bool
+	yesFlag   bool
+)
+
+func NewUpdatePRsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update-prs",
+		Short: "update all PRs generated by the campain",
+		Run:   run,
+	}
+
+	cmd.Flags().BoolVar(&closeFlag, "close", false, "Close all generated PRs")
+	cmd.Flags().BoolVar(&yesFlag, "yes", false, "Skips the confirmation prompt")
+
+	return cmd
+}
+
+// makes sure there is only one action activated
+func onlyOne(args ...bool) bool {
+	// simple counter
+	b := map[bool]int{
+		false: 0,
+		true:  0,
+	}
+	for _, v := range args {
+		b[v] += 1
+	}
+	return b[true] == 1
+}
+
+func validateFlags(closeFlag bool) error {
+	// only option at the moment is `close`
+	if !onlyOne(closeFlag) {
+		return errors.New("update-prs needs one and only one action flag")
+	}
+	return nil
+}
+
+// we keep the args as one of the subfunctions might need it one day.
+func run(c *cobra.Command, args []string) {
+	logger := logging.NewLogger(c)
+	if err := validateFlags(closeFlag); err != nil {
+		logger.Errorf("Error while parsing the flags: %v", err)
+		return
+	}
+
+	if closeFlag {
+		runClose(c, args)
+	}
+}
+
+func runClose(c *cobra.Command, _ []string) {
+	logger := logging.NewLogger(c)
+
+	readCampaignActivity := logger.StartActivity("Reading campaign data")
+	dir, err := campaign.OpenCampaign()
+	if err != nil {
+		readCampaignActivity.EndWithFailure(err)
+		return
+	}
+	readCampaignActivity.EndWithSuccess()
+
+	// Prompting for confirmation
+	if !yesFlag {
+		if !p.AskConfirm(fmt.Sprintf("Close all PRs from the %s campain?", dir.Name)) {
```

Comment:
```suggestion
		if !p.AskConfirm(fmt.Sprintf("Close all PRs from the %s campaign?", dir.Name)) {
```

Reply:
It might be a user-friendly thing if we could say how many PRs will be closed. WDYT?

I'd propose doing this separately, though, so as not to bloat this PR.

Reply:
Totally agree; I added a TODO and will create an issue when this PR is merged.